File: 898-transpose-matrix/transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        a=[]
        for i in range(0,len(matrix[0])):
            b=[]
            for j in range(0,len(matrix)):
                print(j)
                b.append(matrix[j][i])
            a.append(b)
        return a
       
       
       
       
    #    wrong
        # for i in range(0,len(matrix)):
        #     for j in range(0,len(matrix[i])-1):
        #         matrix[i][j],matrix[j][i]=matrix[j][i],matrix[i][j]
        # return matrix
        

File: 898-transpose-matrix/transpose-matrix.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        int rows=matrix[0].size();
        int colums=matrix.size();

        std::vector<std::vector<int>>newMatrix(rows,std::vector<int>(colums,0));

        for(int i=0;i<rows;i++)
        {
            for(int j=0;j<colums;j++)
            {
                newMatrix[i][j]=matrix[j][i];
            }
        }
        return newMatrix;
        
    }
};